astroquery/noirlab/tests/test_noirlab_remote.py

```diff
@@ -38,5 +59,135 @@ def test_query_region_2(self):
+                "original_filename",
+                "instrument",
+                "proc_type"
+            ],
+            "search": [
+                ['original_filename', 'c4d_', 'contains']
+            ]
+        }
+
+        r = Noirlab().query_metadata(qspec, limit=3)
+        actual = r
+        print(f'DBG: test_query_file_metadata={actual.pformat_all()}')
+        expected = exp.query_file_metadata
+        assert actual.pformat_all() == expected
+
+    # ##
+    # ## HDU
+    # ##
+
+    def test_aux_hdu_fields(self):
+        """List the available AUX HDU fields."""
+        r = Noirlab(which='hdu').aux_fields('decam', 'instcal')
+        actual = r
+        print(f'DBG: test_aux_hdu_fields={actual}')
+        expected = exp.aux_hdu_fields
+        assert actual == expected
+
+    def test_core_hdu_fields(self):
+        """List the available CORE HDU fields."""
+        r = Noirlab(which='hdu').core_fields()
+        actual = r
+        print(f'DBG: test_core_hdu_fields={actual}')
+        expected = exp.core_hdu_fields
+        assert actual == expected
+
+    def test_query_hdu_metadata(self):
+        """Search HDU metadata."""
+        qspec = {
+            "outfields": [
+                "fitsfile__archive_filename",
+                "fitsfile__caldat",
+                "fitsfile__instrument",
+                "fitsfile__proc_type",
+                "AIRMASS"  # AUX field. Slows search
+            ],
+            "search": [
+                ["fitsfile__caldat", "2017-08-14", "2017-08-16"],
+                ["fitsfile__instrument", "decam"],
+                ["fitsfile__proc_type", "raw"]
+            ]
+        }
+
+        r = Noirlab(which='hdu').query_metadata(qspec, limit=3)
+        actual = r
+        print(f'DBG: test_query_hdu_metadata={actual.pformat_all()}')
+        expected = exp.query_hdu_metadata
+        assert actual.pformat_all() == expected
+
+    # ##
+    # ## Agnostic
+    # ##
+
+    def test_categoricals(self):
+        """List categories."""
+        r = Noirlab().categoricals()
+        actual = r
+        print(f'DBG: test_categoricals={actual}')
+        expected = exp.categoricals
+        assert actual == expected
+
+    # ##############################################################
+    # ### (3) Other
+    # get_token
+    # retrieve/<md5>
+    # version
+
+    def test_retrieve(self):
+        hdul = Noirlab().retrieve('f92541fdc566dfebac9e7d75e12b5601')
+        actual = list(hdul[0].header.keys())
+        print(f'DBG: test_retrieve={actual}')
+        expected = exp.retrieve
+        assert actual == expected
+
+    def test_version(self):
+        r = Noirlab().version()
+        assert r < 3.0
+
+    def test_get_token(self):
+        actual = Noirlab().get_token('nobody@university.edu', '123456789')
+        expected = {'detail':
+                    'No active account found with the given credentials'}
+        print(f'DBG: test_get_token={actual}')
```

Comment:
remove some, left rest.  Some test for partial results only since its all accessing a live DB.  But "test_categoricals" looks at full results.  They don't change often, but will, then the test will break.  I don't have a good way of handling this except to say that the "exp.categoricals" will have to change in that case.

Reply:
for those asserts checking that the results are at least as many as the expected ones, or looking into that a specific one/subset is in the returned ones would be also acceptable solutions. The don't need to be addressed right now, but can wait until it first diverges.